### PR TITLE
invoke: capture and return stderr if plugin exits unexpectedly

### PR DESCRIPTION
**Pull request by dcbw**
The way raw_exec invokes the command doesn't actually pass back
stderr, despite ExitError having that capability.  c.Run()
internally calls c.Wait() but that doesn't capture stderr and
insert it into the returned ExitError. So we have to do that
ourselves.

Fixes: https://github.com/containernetworking/cni/issues/732
Fixes: https://github.com/containernetworking/cni/pull/759

@bboreham @squeed @mccv1r0 @containernetworking/cni-maintainers

### CODE REVIEW

File: pkg/invoke/raw_exec.go [hunk 1]
```diff
@@ -31,34 +31,40 @@ type RawExec struct {
...
+	c.Stderr = stderr
 	if err := c.Run(); err != nil {
-		return nil, pluginErr(err, stdout.Bytes())
+		return nil, e.pluginErr(err, stdout.Bytes(), stderr.Bytes())
 	}
 
+	// Copy stderr to caller's buffer in case plugin printed to both
+	// stdout and stderr for some reason
+	if e.Stderr != nil && stderr.Len() > 0 {
+		if _, err := stderr.WriteTo(e.Stderr); err != nil {
```
Comment (squeed) [inline pkg/invoke/raw_exec.go:47]: I'd be okay with ignoring this error. Stderr is informational.

File: pkg/invoke/raw_exec.go [hunk 2]
```diff
@@ -31,34 +31,40 @@ type RawExec struct {
...
+			return nil, &types.Error{
+				Msg: fmt.Sprintf("failed to copy netplugin stderr: %v", err),
```
[the remark this answers is not in the document]
Comment (dcbw) [reply pkg/invoke/raw_exec.go]: Decided to ignore it per Casey's comment; but this is a good idea and I should have made it an ErrIOFailure originally.

File: pkg/invoke/raw_exec.go (hunk 1 above)
Comment (dcbw) [reply pkg/invoke/raw_exec.go]: Done.

File: pkg/invoke/raw_exec.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mars1024) [reply pkg/invoke/raw_exec.go]: got it, swallow a non-fatal error is a good choice.

**Pull request merged**